types/ember-animated/animated-each.d.ts

```diff
@@ -0,0 +1,62 @@
+import Component from "@glint/environment-ember-loose/ember-component";
+import AnimatedEachComponent from 'ember-animated/components/animated-each';
+
+// Ideally we'd just get the transition directly from ember-animated
+type Transition = (context: any) => Generator;
+
+export interface AnimatedEachArgs {
+  /** Represents the amount of time an animation takes in miliseconds. */
+  duration?: number;
+
+  /**
+   * When true, all the items in the list will animate as removedSprites when the {{#animated-each}} is destroyed. Defaults to false.
+   *
+   * Note that an <AnimatedOrphans/> component must be actively rendered when this animator is removed for this option to have any effect.
+   */
+  finalRemoval?: boolean;
+
+  /** If set, this animator will only match other animators that have the same group value. */
+  group?: string;
+
+  /** When true, all the items in the list will animate as insertedSprites when the {{#animated-each}} is first rendered. Defaults to false. */
+  initialInsertion?: boolean;
+
+  /** Serves the same purpose as the key in ember {{#each}}, and it's also used to compare values when animating between components. */
+  key?: string;
+
+  /** Specifies data-dependent Rules that choose which Transition to run when the list changes. This takes precedence over use. */
+  rules?: AnimatedEachComponent['rules'];
+
+  /** Specifies the Transition to run when the list changes. */
+  use: Transition;
+
+  /** An optional comma-separated list of properties to observe on each of the objects in the items list. If any of those properties change, we will trigger an animated transition. Without this, we only animate when the list contents change, not when any deeper properties change. */
+  watch: String;
+}
+
+interface AnimatedEachSignature<T> {
+  Args: AnimatedEachArgs & {
+    /* The list of data you are trying to render. */
+    items?: T[]
+  };
+  Yields: {
+    default: [T];
+    else: [];
+  };
+}
+
+export default class AnimatedEach<T> extends Component<AnimatedEachSignature<T>> {}
+
+interface AnimatedEachCurlySignature<T> {
+  Args: AnimatedEachArgs;
+  PositionalArgs: [
+    /* The list of data you are trying to render. */
+    items: T[]
+  ];
+  Yields: {
```

Comment:
@wagenet it seems like we are duplicating the yields here. Thoughts on either making a `AnimatedEachYields` to reduce duplication or do what I did with the `ModifyYields` helper to make a `CurlySignature` util that could be used in the rest of the files as well

```

export type CurlySignature<T, Y> = Omit<T, 'Args'> & Y;

interface AnimatedEachCurlyArgs<T> {
    Args: AnimatedEachArgs;
    PositionalArgs: [
        /* The list of data you are trying to render. */
        items: T[]
    ];
}

export class AnimatedEachCurly<T> extends Component<CurlySignature<AnimatedEachSignature<T>, AnimatedEachCurlyArgs<T>>> {}
```

Reply:
Good idea.